tracer/src/Datadog.Trace/Debugger/Symbols/SymbolBatchUploadApi.cs

```diff
@@ -0,0 +1,133 @@
+// <copyright file="SymbolBatchUploadApi.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+using System;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Datadog.Trace.Agent;
+using Datadog.Trace.Agent.DiscoveryService;
+using Datadog.Trace.Agent.Transports;
+using Datadog.Trace.Debugger.Sink;
+using Datadog.Trace.Logging;
+
+namespace Datadog.Trace.Debugger.Symbols
+{
+    internal class SymbolBatchUploadApi : IBatchUploadApi
+    {
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor<SymbolBatchUploadApi>();
+        private readonly IApiRequestFactory _apiRequestFactory;
+        private readonly ArraySegment<byte> _event;
+        private string? _endpoint;
+
+        private SymbolBatchUploadApi(IApiRequestFactory apiRequestFactory, IDiscoveryService discoveryService)
+        {
+            _apiRequestFactory = apiRequestFactory;
+            _event = GetEventAsArraySegment();
+            discoveryService.SubscribeToChanges(c => _endpoint = c.SymbolDbEndpoint);
+        }
+
+        private ArraySegment<byte> GetEventAsArraySegment()
+        {
+            var @event = string.Empty;
+            try
+            {
+                @event = @$"""ddsource"": ""dd_debugger"",
+""service"": ""{Tracer.Instance.Settings.EnvironmentInternal}""
+""runtimeId"": ""{Tracer.RuntimeId}""";
+
+                var count = Encoding.UTF8.GetByteCount(@event);
+                var eventAsBytes = new byte[count];
+                count = Encoding.UTF8.GetBytes(@event, 0, @event.Length, eventAsBytes, 0);
+                if (count == 0)
+                {
+                    Log.Error("Failed to initialize event {Event}", @event);
+                    return default;
+                }
+
+                return new ArraySegment<byte>(eventAsBytes);
+            }
+            catch (Exception e)
+            {
+                Log.Error(e, "Failed to initialize event {Event}", @event);
+                return default;
+            }
+        }
+
+        public static SymbolBatchUploadApi Create(IApiRequestFactory apiRequestFactory, IDiscoveryService discoveryService)
+        {
+            return new SymbolBatchUploadApi(apiRequestFactory, discoveryService);
+        }
+
+        public async Task<bool> SendBatchAsync(ArraySegment<byte> symbols)
+        {
+            if (Volatile.Read(ref _endpoint) is not { } endpoint)
+            {
+                Log.Warning("Failed to upload symbol: symbol endpoint not yet retrieved from discovery service");
```

Comment:
1. This warning isn't very actionable or helpful to users. I'd suggest something like "Dynamic Instrumentation failed to upload symbols for auto-complete suggestions. Ensure that you are working with datadog-agent v7.45.0 or higher" (am I correct in assuming that's the right version?)
2. We should only write this to log once, and avoid any further attempt to upload symbols until we get a notification from `DiscoveryService` that datadog-agent has been upgraded and the endpoint is now available.